signer/src/block_observer.rs

```diff
@@ -778,11 +779,16 @@ mod tests {
     use crate::testing::block_observer::TestHarness;
     use crate::testing::context::*;
 
+    use rand::rngs::OsRng;
+
     use super::*;
 
     #[test(tokio::test)]
     async fn should_be_able_to_extract_bitcoin_blocks_given_a_block_header_stream() {
-        let mut rng = rand::rngs::StdRng::seed_from_u64(46);
+        let mut rng = OsRng::default();
+        let seed = OsRng::next_u64(&mut rng);
```

Comment:
I believe these can all be simplified to 
```suggestion
        let seed = OsRng.next_u64();
```

Reply:
Alternatively, you can create a test helper function for creating random number generators and printing the seed.